src/main/groovy/com/ullink/AssemblyInfoVersionPatcher.groovy

```diff
@@ -69,6 +71,7 @@ class AssemblyInfoVersionPatcher extends ConventionTask {
     }
 
     void replace(def file, def name, def value) {
+		if (!value || value == '') return
```

Comment:
indent :) also in case the value `0` is valid (is it ?) maybe we could check `== null` instead.
You can play with groovy syntax [here](https://groovyconsole.appspot.com/) also.